src/com/javabasic/lesson6/Apples.java

```diff
@@ -0,0 +1,114 @@
+package com.javabasic.lesson6;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class Apples {
+    static int bigJars;
+    static int middleJars;
+    static int smallJars;
+    static int applesForNextJar;
+
+    static int applesForBigJar = 5;
+    static int applesForMiddleJar = 3;
+
+    static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public static void main(String[] args) throws IOException {
+
+        System.out.println("How many apples do you have?");
+        int intFromUser = getInt();
+
+        getJarsForApples(intFromUser);
+
+        System.out.printf("Big Jars is: %s\n", bigJars);
+        System.out.printf("Middle Jars is: %s\n", middleJars);
+        System.out.printf("Small Jars is: %s", smallJars);
+
+    }
+
+    public static int getInt() throws IOException {
+
+        boolean getInt = false;
+        int intValue = 0;
+
+        while (!getInt) {
+
+            String getValueFromUser = reader.readLine();
+
+            try {
+                intValue = Integer.parseInt(getValueFromUser);
+                if (intValue > 0) {
+                    getInt = true;
+                } else {
+                    System.out.print("Error. Please input an Integer higher than 0: ");
+                }
+
+            } catch (NumberFormatException e) {
+                System.out.print("Incorrect data type. Please input Integer: ");
+            }
+
+        }
+
+        return intValue;
+    }
+
+    public static int[] getCountOfJars(int userInputValue, int maxApplesForJar) {
+
+        int jars = 0;
+        int apples = userInputValue;
+
+        do {
+            jars++;
+            apples -= maxApplesForJar;
+        } while (apples >= maxApplesForJar);
+
+        if (apples == 0) {
+            return new int[]{jars};
+        } else {
+            return new int[]{jars, apples};
+        }
+
+    }
+
+    public static void getJarsForApples(int allApples) {
+
+        if (allApples >= applesForBigJar) {
```

Comment:
This logic is overloaded and has code duplications. 
I propose you optimize it.
Consider using the `/` operation to define how many apples you can put in the jar and the `%` operation to define how many apples rest to distribute.
It can be much simpler, I believe in you))